src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py

```diff
@@ -1380,6 +1362,46 @@ def _aggregate_recharge(
     aggregate_vector = None
 
 
+def _get_monthly_file_lists(n_months, et0_dir_list, precip_dir_list):
+    """Create lists of monthly files for precipitation and evapotranspiration.
+
+    Parameters:
+        n_months (int): The number of months to iterate over (should be 12)
+        et0_dir_list (list): List of file paths in evapotranspiration directory
+        precip_dir_list (list): List of file paths in precipitation directory
+
+    Raises:
+        ValueError: If no file or multiple files are found for a month for
+                    precipitation or et data.
+
+    Returns:
+        tuple: Two lists containing the monthly file paths
+               for evapotranspiration and precipitation, respectively.
+    """
+    et0_path_list = []
+    precip_path_list = []
+    
+    for month_index in range(1, n_months + 1):
+        month_file_pattern = re.compile(r'.*[^\d]0?%d\.[^.]+$' % month_index)
+
+        for data_type, dir_list, path_list in [
+                ('et0', et0_dir_list, et0_path_list),
+                ('Precip', precip_dir_list, precip_path_list)]:
```

Comment:
Now that we have this convenient function to use, what do you think about having the function operate on only one directory, and return one list of filepaths? In other words, instead of using this `for` loop here, we make two separate calls to `_get_monthly_file_list` from up in `execute`.

Reply:
Yes I like this, will update